src/core/ColorPicker/ColorPicker.test.tsx

```diff
@@ -730,3 +820,63 @@ it('should not set focus if setFocus is false', () => {
   expect(container.querySelector('.iui-color-picker')).toBeTruthy();
   expect(container.querySelector('.iui-color-dot')).not.toHaveFocus();
 });
+
+it('should render advanced color picker with opacity slider', () => {
+  const { container } = render(
+    <ColorPicker showAlpha={true}>
+      <ColorBuilder />
+      <ColorInputPanel defaultColorFormat='hsl' />
+    </ColorPicker>,
+  );
+
+  expect(container.querySelector(`.iui-color-picker`)).toBeTruthy();
+  expect(container.querySelector(`.iui-color-selection-wrapper`)).toBeTruthy();
+  expect(
+    container.querySelectorAll(`.iui-color-picker-section-label`).length,
+  ).toBe(1);
+  const element = container.querySelectorAll(
+    `.iui-color-picker-section-label`,
+  )[0];
+  expect(element).toBeDefined();
+  expect(element?.textContent).toBe('HSLA');
+  expect(container.querySelector(`.iui-color-field`)).toBeTruthy();
+  expect(container.querySelector(`.iui-hue-slider`)).toBeTruthy();
+  expect(container.querySelector(`.iui-opacity-slider`)).toBeTruthy();
+  expect(container.querySelector(`.iui-color-dot`)).toBeTruthy();
+  expect(container.querySelectorAll('input').length).toBe(4);
+});
+
+it('should handle arrow key navigation on opacity slider dot', () => {
+  const onSelectionChanged = jest.fn();
+  const selectedColor = ColorValue.create({ h: 0, s: 100, l: 50 });
+  const { container } = render(
+    <ColorPicker
+      showAlpha={true}
+      onChangeComplete={onSelectionChanged}
+      selectedColor={selectedColor}
+    >
+      <ColorBuilder />
+    </ColorPicker>,
+  );
+
+  const colorBuilder = container.querySelector(
+    '.iui-color-picker .iui-color-field',
+  ) as HTMLElement;
+  expect(colorBuilder).toHaveStyle('--hue: #ff0000; --selected-color: #ff0000');
+
+  const opacityDot = container.querySelectorAll(
+    '.iui-slider-thumb',
+  )[1] as HTMLElement;
+  expect(opacityDot).toBeTruthy();
+  expect(opacityDot.style.getPropertyValue('left')).toEqual('100%');
+
+  // Go left
+  fireEvent.keyDown(opacityDot, { key: 'ArrowLeft' });
+  expect(onSelectionChanged).toHaveBeenCalledTimes(1);
+  expect(opacityDot.style.getPropertyValue('left')).toEqual('99%');
+
+  // Go right
+  fireEvent.keyDown(opacityDot, { key: 'ArrowRight' });
+  expect(onSelectionChanged).toHaveBeenCalledTimes(2);
+  expect(opacityDot.style.getPropertyValue('left')).toEqual('100%');
+});
```

Comment:
I think you might also need to test the behavior when showAlpha is false

Reply:
Added test

Reply:
You are only testing the DOM there. You also need to test the color change behavior (i.e. the value returned by `onChangeComplete` should match expectations).

Reply:
Updated tests